lib/Red/AST/Optimizer/OR.pm6

```diff
@@ -0,0 +1,96 @@
+use Red::AST;
+use Red::AST::Infixes;
+use Red::AST::Value;
+unit role Red::AST::Optimizer::OR;
+
+#| x > 1 OR x > 10 ==> x > 10
+multi method optimize(
+    $left  where Red::AST::Ge|Red::AST::Gt,
```

Comment:
I take back the subset from above, I'd create a (possibly empty) role for each set of AST types you declare like this.

Reply:
I prefer to do not leave optimizations info leak for the outside world...